src/builtins/ThresholdPredictors.jl

```diff
@@ -101,8 +173,10 @@ function BinaryThresholdPredictor(args...;
         atom = model
     end
 
-    metamodel =
-        _type_given_atom[MMI.abstract_type(atom)](atom, Float64(threshold))
+    A = MMI.abstract_type(atom)
+    T = get(_type_given_atom, A, nothing)
+    isnothing(T) && throw(err_unsupported_model_type(A))
```

Comment:
```suggestion
   T = threshold_constructor(A)
```

Then somewhere below/above add the following
```julia
for (atom, type) in type_given_atom
    @eval threshold_constructor(::Type{$atom}) = $type
end
threshold_constructor(A) = throw(err_unsupported_model_type(A))

This approach is type stable.

Reply:
Done.